src/unity/python/turicreate/data_structures/sarray.py

```diff
@@ -2860,6 +2860,14 @@ def show(self, title=None, xlabel=None, ylabel=None):
         import os
         (tcviz_dir, _) = os.path.split(os.path.dirname(__file__))
         path_to_client = os.path.join(tcviz_dir, 'Turi Create Visualization.app', 'Contents', 'MacOS', 'Turi Create Visualization')
+
+        if title is "":
```

Comment:
I think you may need this logic in [show.py](https://github.com/apple/turicreate/blob/master/src/unity/python/turicreate/visualization/show.py#L72) as well?

Reply:
Updated with requested changes

Reply:
Actually I think you want to use `==` instead of `is` for string comparison; see [this post](https://stackoverflow.com/questions/1504717/why-does-comparing-strings-in-python-using-either-or-is-sometimes-produce).